### ISSUE TITLE
Dev Home Environments list is duplicating results

### ISSUE THREAD

**Issue by acangialosi** [microsoft/devhome#3591]
### Dev Home version

0.1601.561.0

### Windows build number

10.0.22635.4010

### Other software

OS Build Version: 10.0.22635.4010.amd64fre.ni_release.220506-1250
.NET Version: .NET 8.0.6

### Steps to reproduce the bug

Start Dev home
Open Environments tab

### Expected result

Show only one instance per dev box.

### Actual result

Environments list is duplicating results. 

![image](https://github.com/user-attachments/assets/413b8b20-fb8c-4b21-82ba-a8fe9e3c2201)

### Included System Information

CPU: Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
Physical Memory: 31.6GB (14.42GB free)
Processor Architecture: x64

### Included Extensions Information

Extensions: 
Microsoft.Windows.DevHome.Canary_0.1401.500.0_x64__8wekyb3d8bbwe (Core Widget Extension)
Microsoft.Windows.DevHome.Canary_0.1401.500.0_x64__8wekyb3d8bbwe (Hyper-V Extension)
Microsoft.Windows.DevHomeAzureExtension.Canary_0.1000.549.0_x64__8wekyb3d8bbwe (Dev Home Azure Extension (Canary))
Microsoft.Windows.DevHome_0.1601.561.0_x64__8wekyb3d8bbwe (Core Widget Extension)
Microsoft.Windows.DevHome_0.1601.561.0_x64__8wekyb3d8bbwe (Hyper-V Extension)
Microsoft.Windows.DevHome_0.1601.561.0_x64__8wekyb3d8bbwe (Windows Subsystem for Linux Extension)
Microsoft.Windows.DevHomeAzureExtension_0.1100.561.0_x64__8wekyb3d8bbwe (Dev Home Azure Extension (Preview))

Widget Service: 
MicrosoftWindows.Client.WebExperience_524.21802.0.0_x64__cw5n1h2txyewy

**Comment by bbonaby** [comment]
Hi @acangialosi  I have a similar comment to the one I made in the issue about the pools not being sorted alphabetically. in Issue #3592 . It looks like on your computer both the Preview version of the Azure extension and the Canary version are installed. I just tested out the latest version of Dev Home: ver 0.1601.561.0 and the Azure extension ver: 0.1100.561.0  and confirmed that the list of DevBoxes aren't duplicated. You likely will need to disable/uninstall the canary version and then only one set will appear. Let me know if that fixes your issue. Thanks.
